frontend/tests/controllers/ProblemCreateTest.php

```diff
@@ -849,44 +789,48 @@ public function testCreateProblemWithDefaultValues() {
         }
     }
 
-    public function testCreateProblemWithDifferentShowDiffValues() {
-        $showDiffValues = ['none', 'examples', 'all', 'invalid'];
+    /**
+     * @dataProvider showDiffValueProvider
+     */
+    public function testCreateProblemWithValidShowDiffValues(
+        string $showDiffValue
+    ) {
+        $problemData = \OmegaUp\Test\Factories\Problem::getRequest(
+            new \OmegaUp\Test\Factories\ProblemParams([
+                'show_diff' => $showDiffValue,
+            ])
+        );
+        $r = $problemData['request'];
+        $problemAuthor = $problemData['author'];
 
-        foreach ($showDiffValues as $showDiffValue) {
-            $problemData = \OmegaUp\Test\Factories\Problem::getRequest(
-                new \OmegaUp\Test\Factories\ProblemParams([
-                    'show_diff' => $showDiffValue,
-                ])
-            );
-            $r = $problemData['request'];
-            $problemAuthor = $problemData['author'];
+        // Login user
+        $login = self::login($problemAuthor);
+        $r['auth_token'] = $login->auth_token;
 
-            // Login user
-            $login = self::login($problemAuthor);
-            $r['auth_token'] = $login->auth_token;
+        \OmegaUp\Controllers\Problem::apiCreate($r);
+        $problem = \OmegaUp\DAO\Problems::getByAlias($r['problem_alias']);
```

Comment:
según yo debería ser posible reemplazar todo esto por

```suggestion
        [
            'problem' => $problem,
        ] = \OmegaUp\Test\Factories\Problem::createProblem(
            new \OmegaUp\Test\Factories\ProblemParams([
                'show_diff' => $showDiffValue,
            ])
        );
```

Reply:
Listo